cmd/upgrade/machinepool/cmd.go

```diff
@@ -369,20 +367,10 @@ func ComputeNodePoolVersion(r *rosa.Runtime, cmd *cobra.Command, cluster *cmv1.C
 	return version, nil
 }
 
-func GetAvailableVersion(r *rosa.Runtime, cluster *cmv1.Cluster, nodePool *cmv1.NodePool) ([]string, error) {
-	clusterVersion := cluster.Version().RawID()
-	nodePoolVersion := ocm.GetRawVersionId(nodePool.Version().ID())
-	// This is called in HyperShift, but we don't want to exclude version which are HCP disabled for node pools
-	// so we pass the relative parameter as false
-	versionList, err := versions.GetVersionList(r, cluster.Version().ChannelGroup(), true, false, false)
-	if err != nil {
-		return nil, err
-	}
+func formatNodePool(nodePool *cmv1.NodePool) string {
+	var nodePoolJson bytes.Buffer
 
-	// Filter the available list of versions for a hosted machine pool
-	filteredVersionList := versions.GetFilteredVersionListForUpdate(versionList, nodePoolVersion, clusterVersion)
-	if err != nil {
-		return nil, err
-	}
-	return filteredVersionList, nil
+	cmv1.MarshalNodePool(nodePool, &nodePoolJson)
```

Comment:
Mmm this is weird, using the SDK, why do you need to perform a marshall at this point?

(perhaps I missed context in the PR)

Reply:
good catch, this is an helper function for the tests, but I have now a generic helper so I remove it in favour of the generic method. Nothing for the normal flow in any case